### PR TITLE
docs: added more functions under rest API docs section

### PR DESCRIPTION
**Pull request by audrium**
Newly added functions:
- upload_to_server
- list_files
- get_workflow_status

Also, tried to make parameters descriptions more consistent

### CODE REVIEW

File: reana_client/api/client.py [hunk 1]
```diff
@@ -491,9 +496,10 @@ def upload_to_server(workflow, paths, access_token):
 
     Shared e.g. by `code upload` and `inputs upload`.
 
-    :param workflow: ID of that workflow whose workspace should be
+    :param workflow: name or id of workflow whose workspace should be
         used to store the files.
     :param paths: Absolute filepath(s) of files to be uploaded.
```
Comment (mvidalgarcia) [inline reana_client/api/client.py:501]:
```suggestion
    :param paths: absolute filepath(s) of files to be uploaded.
```

File: reana_client/api/client.py [hunk 2]
```diff
@@ -369,19 +371,19 @@ def get_workflow_logs(workflow_id, access_token, steps=None, page=None, size=Non
         raise e
 
 
-def download_file(workflow_id, file_name, access_token):
+def download_file(workflow, file_name, access_token):
     """Download the requested file if it exists.
 
-    :param workflow_id: UUID which identifies the workflow.
+    :param workflow: name or id which identifies the workflow.
     :param file_name: File name or path to the file requested.
```
Comment (mvidalgarcia) [inline reana_client/api/client.py:378]:
```suggestion
    :param file_name: file name or path to the file requested.
```

**Review by mvidalgarcia — APPROVED** [review]
LGTM 👍

**Comment by tiborsimko** [comment]
> PS. Should we point to `maint-0.7`?

Now available on RTFD:  https://reana-client.readthedocs.io/en/maint-0.7/

**Pull request merged**